tests/Support/SupportCollectionTest.php

```diff
@@ -895,6 +895,35 @@ public function testFlatMap()
         $this->assertEquals(['programming', 'basketball', 'music', 'powerlifting'], $data->all());
     }
 
+    public function testMapToAssoc()
+    {
+        $data = new Collection([
+            ['name' => 'Blastoise', 'type' => 'Water', 'idx' => 9],
+            ['name' => 'Charmander', 'type' => 'Fire', 'idx' => 4],
+            ['name' => 'Dragonair', 'type' => 'Dragon', 'idx' => 148],
+        ]);
+        $data = $data->mapToAssoc(function ($pokemon) {
+            return [$pokemon['name'] => $pokemon['type']];
+        });
+        $this->assertEquals(
+            ['Blastoise' => 'Water', 'Charmander' => 'Fire', 'Dragonair' => 'Dragon'],
+            $data
+        );
+    }
+
+    public function testToAssoc()
+    {
+        $collection = new Collection([
+            ['Blastoise' => 'Water'],
+            ['Charmander' => 'Fire'],
+            ['Dragonair' => 'Dragon'],
+        ]);
+        $this->assertEquals(
+            ['Blastoise' => 'Water', 'Charmander' => 'Fire', 'Dragonair' => 'Dragon'],
+            $collection->toAssoc()
+        );
+    }
```

Comment:
I think it would behave differently with numeric keys, but it probably just shouldn't be used with numeric keys.

I think it's fine to have a more explicitly named method for this purpose, even if under the hood it was just delegating to collapse, if only for expressiveness and to make it easier for the implementations to diverge if they ever need to.

We might be able to just keep `mapToAssoc` though, this method might not actually get used outside of that? Another idea is to make this smart enough to handle `[key => value]` _or_ `[key, value]`, but might just be trying to justify its existence.